src/passes/CodeFolding.cpp

```diff
@@ -304,13 +304,28 @@ struct CodeFolding : public WalkerPass<ControlFlowWalker<CodeFolding>> {
         // anything exiting that is in all targets is something bad
         return false;
       }
-      // Currently pop instructions are only used for exnref.pop, which is a
-      // pseudo instruction following a catch. We check if the current
-      // expression has a pop child. This can be overly conservative, because
-      // this can also exclude whole try-catches that contain a pop within them.
-      if (getModule()->features.hasExceptionHandling() &&
-          !FindAll<Pop>(item).list.empty()) {
-        return false;
+      if (getModule()->features.hasExceptionHandling()) {
+        // Currently pop instructions are only used for exnref.pop, which is a
+        // pseudo instruction following a catch. We check if the current
+        // expression has a pop child. This can be overly conservative, because
+        // this can also exclude whole try-catches that contain a pop within
+        // them.
+        if (!FindAll<Pop>(item).list.empty()) {
```

Comment:
I think we recently added a `contains` method? Or did that not land yet?

Reply:
It didn't land yet, but I deleted the `depth = -1` option that enables search of all children anyway, because we can use `FindAll` for that instead.